src/WebJobs.Script.Analyzers/DiagnosticDescriptors.cs

```diff
@@ -27,5 +27,11 @@ private static DiagnosticDescriptor Create(string id, string title, string messa
                 messageFormat: "Reuse HttpClient instances to avoid holding more connections than necessary. See helplink for more information.",
                 category: Constants.DiagnosticsCategories.Reliability,
                 severity: DiagnosticSeverity.Warning);
+
+        public static DiagnosticDescriptor IllegalFunctionName { get; }
+            = Create(id: "AZF0003", title: "Illegal Function name",
+                messageFormat: "Function name can't be '{0}'",
```

Comment:
Is there any reasonable way for us to explain the regex to provide clues? Like "cannot be longer than 32 characters and must be alphanumeric?"

Reply:
How about: `Function name can't be '{0}'. It must start with a letter, be no longer than 128 characters, and contain only alphanumeric characters, underscores, or hyphens.`

128-character limit: https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs/FunctionNameAttribute.cs#L35

What logic imposes a 32-character limit at runtime? AFAIK we do this for host IDs, but I wasn't aware we imposed the same limit for function names.

Reply:
That sounds good to me!

The 32-character limit is for host ids b/c we use that to create queues for some internal stuff, which have stricter limits on length. That logic is here -- https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Host/ScriptHostIdProvider.cs#L87